console/src/components/Services/Data/TablePermissions/molecules/PermButtonsSection.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback } from 'react';
+import Button from '../../../../Common/Button';
+import {
+  isJsonString,
+  getConfirmation,
+} from '../../../../Common/utils/jsUtils';
+import { FilterState } from '../utils';
+import { showErrorNotification } from '../../../Common/Notification';
+import { permChangePermissions, permChangeTypes } from '../Actions';
+import styles from '../../../../Common/Permissions/PermissionStyles.scss';
+
+interface PermButtonSectionProps {
+  readOnlyMode: string;
+  query: string;
+  localFilterString: FilterState;
+  dispatch: (...p: any) => void;
+  permissionsState: FilterState;
+  permsChanged: string;
+  currQueryPermissions: string;
+}
+
+const PermButtonSection: React.FC<PermButtonSectionProps> = ({
+  readOnlyMode,
+  query,
+  localFilterString,
+  dispatch,
+  permissionsState,
+  permsChanged,
+  currQueryPermissions,
+}) => {
+  if (readOnlyMode) {
+    return null;
+  }
+
+  const dispatchSavePermissions = useCallback(() => {
```

Comment:
`useCallback` is a good solution when a component gets many props but a function is dependent only on a few. In this case, you're passing most of them (`readOnlyMode, query, localFilterString, dispatch`), so of course, it's not bad, it can stay, but there's no much value in that.